Dafny 4: edits to Refinement and Attributes sections (#3581)
dafny-lang/dafny: docs/DafnyRef/Attributes.md

It can take one of the following forms:

* `{:nativeType}` - With no parameters it has no effect and the declaration
will have its default behavior, which is to choose a native type that can hold any

> value satisfying the constraints, if possible, otherwise BigInteger is used.

Reviewer: ```suggestion
value satisfying the constraints, if possible, and otherwise to use BigInteger.
```
